test/adt_test.exs

```diff
@@ -61,4 +61,12 @@ defmodule AdtTest do
       _ -> false
     end)
   end
+
+  test "adtcase allows per-module pattern matching" do
+    foo = %AdtDefinition.Foo{}
+    assert 0 == (ADT.case AdtDefinition, foo do
```

Comment:
Just an idea, maybe each ADT module could contain a `case` function:

```elixir
AdtDefinition.case foo do
# ...
end
```

So you wouldn't need to select the ADT module in the params